course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala

```diff
@@ -419,6 +422,82 @@ class CourseEnrolmentActor @Inject()(@Named("course-batch-notification-actor") c
+                enrollProgramCourses(request)
+            } else {
+                ProjectCommonException.throwClientErrorException(ResponseCode.contentTypeMismatch, courseId)
+            }
+        }
+    }
+
+    def enrollProgramCourses(request: Request): Unit = {
+        try {
+            val courseId: String = request.get(JsonKey.COURSE_ID).asInstanceOf[String]
+            val userId: String = request.get(JsonKey.USER_ID).asInstanceOf[String]
+            val contentData = ContentUtil.getContentReadV2(courseId, request.getContext.getOrDefault(JsonKey.HEADER, new util.HashMap[String, String]).asInstanceOf[util.Map[String, String]])
+            if (contentData.size() == 0)
+                ProjectCommonException.throwClientErrorException(ResponseCode.accessDeniedToEnrolOrUnenrolCourse, courseId)
+            val batchesForCourse: java.util.List[java.util.Map[String, AnyRef]] = contentData.get(JsonKey.BATCHES).asInstanceOf[java.util.List[java.util.Map[String, AnyRef]]]
+            val filteredBatches = batchesForCourse.filter(batch => batch.get(JsonKey.STATUS) != 2).toList
+            val batchId: String = filteredBatches.get(0).get(JsonKey.BATCH_ID).asInstanceOf[String]
+            val batchData: CourseBatch = courseBatchDao.readById(courseId, batchId, request.getRequestContext)
+            val enrolmentData: UserCourses = userCoursesDao.read(request.getRequestContext, userId, courseId, batchId)
+            val batchUserData: BatchUser = batchUserDao.read(request.getRequestContext, batchId, userId)
+            validateEnrolment(batchData, enrolmentData, true)
+            val dataBatch: util.Map[String, AnyRef] = createBatchUserMapping(batchId, userId, batchUserData)
+            val data: java.util.Map[String, AnyRef] = createUserEnrolmentMap(userId, courseId, batchId, enrolmentData, request.getContext.getOrDefault(JsonKey.REQUEST_ID, "").asInstanceOf[String])
+            upsertEnrollment(userId, courseId, batchId, data, dataBatch, (null == enrolmentData), request.getRequestContext)
+            logger.info(request.getRequestContext, "CourseEnrolmentActor :: enroll :: Deleting redis for key " + getCacheKey(userId))
+            cacheUtil.delete(getCacheKey(userId))
+            sender().tell(successResponse(), self)
+            generateTelemetryAudit(userId, courseId, batchId, data, "enrol", JsonKey.CREATE, request.getContext)
+            notifyUser(userId, batchData, JsonKey.ADD)
+        } catch {
+            case e: ProjectCommonException =>
+                logger.error(request.getRequestContext, "Exception in upsertEnrollment list : user ::" + e.getMessage, e)
+                if (e.getMessage.equals(ResponseCode.userAlreadyEnrolledCourse.getErrorMessage))
+                    return;
+                if (e.getMessage.equals(ResponseCode.courseBatchAlreadyCompleted.getErrorMessage))
+                    return;
+                if (e.getMessage.equals(ResponseCode.courseBatchEnrollmentDateEnded.getErrorMessage))
```

Comment:
it must be 
ResponseCode.courseBatchEnrollmentDateEnded.getErrorMessage.equals(e.getMessage)
We should check against the constant... not against variables

Reply:
Done and make changes